xonsh/main.py

```diff
@@ -386,6 +372,20 @@ def func_sig_ttin_ttou(n, f):
             code = sys.stdin.read()
             run_code_with_cache(code, shell.execer, glb=shell.ctx, loc=None,
                                 mode='exec')
+        if args.force_interactive:
```

Comment:
Yeah, I just tried it.

```console
scopatz@artemis:~$ xonsh -i -c "echo hi"   # in bash
hi
scopatz@artemis ~ $  exit # in xonsh 
scopatz@artemis:~$  # back in bash
```

We should never enter into xonsh fully. Take the inverse, for example,

```console
scopatz@artemis ~/xonsh/news nullsub $ bash -i -c "echo hello"  # in xonsh
bash: cannot set terminal process group (-1): Inappropriate ioctl for device
bash: no job control in this shell
hello
scopatz@artemis ~/xonsh/news interactive $ # still in xonsh
```

Does this make sense?